lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -98,23 +109,30 @@ public synchronized void setRunning() {
   }
 
   /**
-   * @return idle, if there are no outstanding requests or allocations. Not idle else.
+   * Driver is idle if, regardless of status, it has no evaluators allocated and no pending container requests.
+   * @return true if the driver can be considered idle, false otherwise.
```

Comment:
Good point, but I am not sure that a private method would be a good place for that piece of documentation. I would rather add it to the `DriverIdlenessSource` interface or close to the code that uses it, i.e. `DriverIdleManager` or `DriverStatusManager`